static/test/javascripts/spec/commercial/hosted/next-video-autoplay.spec.js

```diff
@@ -0,0 +1,94 @@
+define([
+    'common/utils/$',
+    'helpers/fixtures',
+    'Promise',
+    'fastdom',
+    'helpers/injector'
+], function (
+    $,
+    fixtures,
+    Promise,
+    fastdom,
+    Injector
+) {
+    var nextVideoAutoplay,
+        injector = new Injector();
+
+    describe('Next video autoplay', function () {
+
+        var fixturesConfig = {
+                id: 'next-video-autoplay',
+                fixtures: [
+                    '<div><video data-duration="160">' +
+                    '<source type="video/mp4" src="">' +
+                    '</video></div>' +
+                    '<div class="js-hosted-next-autoplay">' +
+                    '<div class="js-autoplay-timer" data-next-page="/commercial/advertiser-content/renault-car-of-the-future/design-competition-episode2">10s</div>' +
+                    '</div>' +
+                    '<button class="js-autoplay-cancel">' +
+                    '</button>'
+                ]
+            },
+            $fixturesContainer;
+
+        beforeEach(function (done) {
+            injector.mock('common/modules/analytics/google', function noop() {});
+            injector.mock('commercial/modules/hosted/next-video',  {
+                load: Promise.resolve
+            });
+            injector.require([
+                'commercial/modules/hosted/next-video-autoplay'
+            ], function () {
+                nextVideoAutoplay = arguments[0];
+
+                $fixturesContainer = fixtures.render(fixturesConfig);
+                nextVideoAutoplay.init().then(done);
+            });
+        });
+
+        afterEach(function () {
+            fixtures.clean(fixturesConfig.id);
+        });
+
+        it('should exist', function (done) {
+            expect(nextVideoAutoplay).toBeDefined();
+            done();
+        });
+
+        it('should trigger autoplay when there is a next video', function (done) {
+            expect(nextVideoAutoplay.canAutoplay()).toBeTruthy();
+            done();
+        });
+
+        it('should show end slate information', function (done) {
+            nextVideoAutoplay.triggerEndSlate();
+            fastdom.defer(1, function () {
```

Comment:
This is fine for now, but when we upgrade Fastdom, `fastdom.defer` has been deprecated. You can just call `fastdom.read`, or use `fastdom-promise` in your module.